DataStructures/Graphs/BFSAdjacencyListGraph.java

```diff
@@ -0,0 +1,156 @@
+
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Adjacency List implementation of Graph
+ * @author Abhishek
+ *
+ */
+public class BFSAdjacencyListGraph {
+	LinkedList<Vertex> adjacencyList;
+
+	public BFSAdjacencyListGraph() {
+		adjacencyList = new LinkedList<Vertex>();
+	}
+
+	/**
+	 * Add vertex in the graph.
+	 * 
+	 * @param vertex vertex instance
+	 */
+	public void addVertex(Vertex vertex) {
+		adjacencyList.add(vertex);
+	}
+
+	/**
+	 * Add neighbor for the vertex at vertexIndex in the adjacency list.
+	 * 
+	 * @param vertexIndex   - index of vertex where neighbor to be added.
+	 * @param neighborIndex - index of neighboring vertex.
+	 */
+	public void addNeighbor(int vertexIndex, int neighborIndex) {
+		adjacencyList.get(vertexIndex).addNeighbor(neighborIndex);
+	}
+
+	/**
+	 * BFS all the connected vertices.
+	 *
+	 * @param startVertex starting vertex
+	 * @param visited     store all the visited vertices
+	 */
+	private void bfs(int startVertex, boolean[] visited) {
+		Queue<Integer> queue = new LinkedList<Integer>();
+		queue.add(startVertex);
+		while (!queue.isEmpty()) {
+			int vertexIndex = queue.poll();
+			if (!visited[vertexIndex]) {
+				Vertex vertex = adjacencyList.get(vertexIndex);
+				System.out.print(vertex.getVertexName() + "-->");
+				visited[vertexIndex] = true;
+				// Traverse all neighbors and add to queue
+				for (int neighborIndex : vertex.getNeighbours()) {
+					queue.add(neighborIndex);
+				}
+			}
+		}
+	}
+
+	/**
+	 * BFS Runner for traversing all the disconnected vertices.
+	 */
+	public void bfs() {
+		boolean[] visited = new boolean[adjacencyList.size()];
+		for (int i = 0; i < visited.length; i++) {
+			if (visited[i] == false) {
+				bfs(i, visited);
+			}
+		}
+	}
+
+	/**
+	 * Main method.
+	 * 
+	 * @param args command line arguments
+	 */
+	public static void main(String[] args) {

```

Comment:
If possible add test class instead of using `main` method. 

If you don't know how to write test class, please refer to [DecimalToAnyBaseTest.java](https://github.com/TheAlgorithms/Java/blob/Development/src/test/java/com/conversions/DecimalToAnyBaseTest.java)